assists/domain_access.py

```diff
@@ -40,6 +40,79 @@ def check_bit_set(n, k):
 
     return False
 
+def node_memories(tree, node):
+    memory_nodes = []
+    try:
+        memory_list = node["memory"].value
+    except:
+        memory_list = []
+
+    if memory_list:
+        memory_nodes.append(node)
+
+    return memory_nodes
+
+def get_domain_nodes(tree):
+    try:
+        domain_node = tree['/domains']
+    except:
+        domain_node = None
+    direct_node_refs = []
+
+    if domain_node:
+        prop_dict = domain_node.child_nodes
```

Comment:
This really isn't a "prop_dict", but are the children.

Also, we should use the subnodes() routine where possible, since it allows the underlying tree structure to be adjusted in the future.